doc/user/content/sql/alter-cluster.md

```diff
@@ -47,6 +60,28 @@ See the reference documentation for [`CREATE CLUSTER`](../create-cluster/#schedu
 or [`CREATE MATERIALIZED VIEW`](../create-materialized-view/#refresh-strategies)
 for more details on scheduled clusters.
 
+### No-downtime reconfiguration
+
+{{< private-preview />}}
+Changing the configuration of a cluster using the `ALTER CLUSTER` command
+requires the cluster to restart, which incurs **downtime**. For clusters that
+don't contain sources or sinks, you can use the `WAIT UNTIL READY` option to
+perform a graceful reconfiguration, which spins up an additional cluster
+replica under the covers with the desired new configuration, waits for the
+replica to be hydrated, and then replaces the original replica. This allows you
+to perform operations like cluster resizing with **no downtime**.
+
+```sql
+ALTER CLUSTER c1 SET (SIZE '100CC') WITH (WAIT UNTIL READY (TIMEOUT = '10m', ON TIMEOUT = 'COMMIT'))
+````
+
+The `ALTER` statement will block while the new replica becomes ready, which
+could take as long as the specified timeout. During this operation, any other
+reconfiguration command will block, and any connection interruption or
+statement cancelation will cause a rollback — no configuration changes will
+take effect in that case.
```

Comment:
QQ: Is it that the statement blocks all? or just other reconfiguration commands?  If first, is it more (feel free to ignore suggestion)

```suggestion
The `ALTER CLUSTER` statement will block any other reconfiguration commands
while the new replica becomes ready. This could take as long as the specified
timeout. During the operation, any connection interruption or statement
cancelation will cause a rollback; i.e., no configuration changes from the
statement will take effect.
```

Reply:
Interesting... both?

The statement will block as in it will not return until the command has succeeded, but it will also prevent reconfiguration of the cluster while it's executing.

Reply:
I also updated the text here to spell that out a little more clearly.